app/controllers/api/base_controller/renderer.rb

```diff
@@ -569,6 +569,18 @@ def render_create_resource_options(ems_id)
         render_options(type, :form_schema => klass.params_for_create(ems))
       end
 
+      def render_attach_resource_options(id)
+        type = @req.collection.to_sym
+        resource = resource_search(id, type)
+        raise BadRequestError, resource.unsupported_reason(:attach_volume) unless resource.supports?(:attach_volume)
+
+        if resource.try(:params_for_attach) ## TODO can we make this one line?
+          render_options(type, :form_schema => resource.params_for_attach)  
+        else
+          render_options(type, :form_schema => {}) ## {:fields => []}
+        end
```

Comment:
```suggestion
        render_options(type, :form_schema => resource.try(:params_for_attach) || {})
```

Reply:
I think that should get you to a single line as per your TODO.

Reply:
Ill try this change in my other PR where I keep render_options https://github.com/ManageIQ/manageiq-api/pull/1141/files#diff-48fb0a597ebc41e5c8c4fde22100d5577a08a07b84761152f69eaa2f580c4c37R565-R569